src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs

```diff
@@ -565,7 +565,7 @@ private static unsafe int CreateArrayGCDesc(LowLevelList<bool> bitfield, int ran
             {
                 if (numSeries > 0)
                 {
-                    *ptr-- = (short)((bitfield.Count - last) * IntPtr.Size);
+                    *ptr-- = (short)((first + bitfield.Count - last) * IntPtr.Size);
```

Comment:
Right. The skip after the last pointer in an array element is not just to the end of the element, it is to the first pointer in the next element. 

This will matter when the first pointer in the element type is not at 0. It is rare, since the auto layout puts pointers to the start of the struct, but definitely possible - when the fields are not sorted or when structs are nesting.

This is a very good catch!! Needs to be upstreamed.

Reply:
Native `CreateArrayMethodTable` also adds `pElemMT->GetBaseSize() - OBJECT_BASESIZE`, although I do not think a struct can have a base size that is not == `OBJECT_BASESIZE`.

https://github.com/dotnet/runtime/blob/9915e9825edf174df73bc6db9ab59f167e92ab3c/src/coreclr/vm/array.cpp#L623-L624